tests/Main.hs

```diff
@@ -0,0 +1,67 @@
+import Data.Proxy       (Proxy (..))
+import System.Directory (doesFileExist)
+import System.FilePath  (takeDirectory, FilePath, (</>))
+import Test.Tasty
+import Test.Tasty.Golden  (findByExtension)
+import Test.Tasty.Options (IsOption (..), OptionDescription (Option))
+import Test.Tasty.Program (testProgram)
+
+
+main :: IO ()
+main = do
+  input <- testInput
+  defaultMainWithIngredients ings . askOption $ \args -> mkTestGrp args input
+    where
+      ings = includingOptions [Option (Proxy :: Proxy Args)] :
+        defaultIngredients
+
+newtype Args = A String
+
+instance IsOption Args where
+  defaultValue = A ""
+  parseValue = Just . A
+  optionName = return "sixten-args"
+  optionHelp = return "Arguments to supply to the sixten binary."
+
+testRootDir :: FilePath
+testRootDir = "tests"
+testName :: FilePath -> FilePath
+testName = drop (length testRootDir + 1)
+
+groupByKey :: Ord b => (a -> b) -> [a] -> [(b, [a])]
+groupByKey _  [] = []
+groupByKey f (x : xs) = (kx, x : xlike) : groupByKey f rest
+  where kx = f x
+        (xlike, rest) = span ((==) kx . f) xs
+
+testInput :: IO [(String, [String], Maybe String)]
```

Comment:
What's the `Maybe String` for? If I'm reading the code correctly it's always `Nothing`. Considering removing it if that's the case.

Reply:
I kept it in case you need to put in some `stdin` argument later, you only need to modify the `testInput` function and not the `mkTestGrp` function. I can remove it, no big deal.